diagrams/lib/transitions.dart

```diff
@@ -0,0 +1,599 @@
+    _diagrams.add(const DecoratedBoxTransitionDiagram());
+    _diagrams.add(const FadeTransitionDiagram());
+    _diagrams.add(const PositionedTransitionDiagram());
+    _diagrams.add(const RelativePositionedTransitionDiagram());
+    _diagrams.add(const RotationTransitionDiagram());
+    _diagrams.add(const ScaleTransitionDiagram());
+    _diagrams.add(const SizeTransitionDiagram());
+    _diagrams.add(const SlideTransitionDiagram());
+  }
+
+  final List<TransitionDiagram<dynamic>> _diagrams = <TransitionDiagram<dynamic>>[];
+
+  @override
+  final String category = 'widgets';
+
+  @override
+  Future<List<DiagramMetadata>> get diagrams async => _diagrams;
+
+  @override
+  Future<File> generateDiagram(DiagramMetadata diagram) async {
+    final TransitionDiagram<dynamic> typedDiagram = diagram;
+    controller.builder = (BuildContext context) => typedDiagram;
+
+    final Map<Duration, DiagramKeyframe> keyframes = <Duration, DiagramKeyframe>{
+      Duration.zero: (Duration now) async {
+        final RenderBox target = _transitionKey.currentContext.findRenderObject();
+        final Offset targetOffset = target.localToGlobal(target.size.center(Offset.zero));
+        final TestGesture gesture = await controller.startGesture(targetOffset);
+        await gesture.up();
+      },
+      const Duration(seconds: 3): (Duration now) async {
+        final RenderBox target = _transitionKey.currentContext.findRenderObject();
+        final Offset targetOffset = target.localToGlobal(target.size.center(Offset.zero));
+        final TestGesture gesture = await controller.startGesture(targetOffset);
+        await gesture.up();
+      },
+    };
+
+    final File result = await controller.drawAnimatedDiagramToFiles(
+      end: _kOverallAnimationDuration,
+      frameRate: _kAnimationFrameRate,
+      name: diagram.name,
+      category: category,
+      keyframes: keyframes,
+    );
+    return result;
+  }
+}
+
+abstract class TransitionDiagram<T> extends StatefulWidget implements DiagramMetadata {
+  const TransitionDiagram({
+    Key key,
+  }) : super(key: key);
+
+  /// The animation curve for both the animation and the sparkline to use.
+  Curve get curve;
+  Animation<T> buildAnimation(AnimationController controller);
+}
+
+abstract class TransitionDiagramState<T> extends State<TransitionDiagram<T>> //
+    with
+        TickerProviderStateMixin<TransitionDiagram<T>> {
+  bool selected = false;
+  Animation<T> animation;
+  AnimationController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = new AnimationController(
+      duration: _kAnimationDuration,
+      vsync: this,
+    )..addListener(() {
+        setState(() {});
+      });
+    animation = widget.buildAnimation(_controller);
+  }
+
+  @override
+  void dispose() {
+    _controller?.dispose();
+    super.dispose();
+  }
+
+  Widget wrap(BuildContext context, Widget child) {
+    return GestureDetector(
```

Comment:
i think we're still giving the "new", though the lint to stop that is coming soon i hope